_posts/2023-06-23-run-and-debug-rust-verification-harnesses-natively-on-vscode.md

```diff
@@ -0,0 +1,149 @@
+---
+layout: post
+title:  "Run and debug Rust verification harnesses natively on VS Code"
+---
+
+Kani is a verification tool that can help you prove properties about your Rust code. To learn more about Kani, check out the [Kani tutorial](https://model-checking.github.io/kani/kani-tutorial.html) and our [previous blog posts](https://model-checking.github.io/kani-verifier-blog/).
+
+In today’s blog post, we introduce the [Kani VS Code Extension](https://marketplace.visualstudio.com/items?itemName=model-checking.kani-vscode-extension), available on the VS Code marketplace. The extension automatically detects all harness within the project, and allows users to run and debug them directly in VS Code. To allow an easier verification experience, Kani is now usable directly from the VS Code user interface. Until now, developers could only run and debug harnesses via a command-line interface.
+
+<img src="{{site.baseurl | prepend: site.url}}/assets/images/vs-code-images/kani-demo.gif" style="box-shadow: 0 2px 4px rgba(0, 0, 0, 0.5);" alt="view-kani-demo" />
+
+## A Simple Example
+
+To show some of the features in the extension, we will work through a familiar example, the rectangle example that we introduced in our [first blog post](https://model-checking.github.io/kani-verifier-blog/2022/05/04/announcing-the-kani-rust-verifier-project.html).
+
+```rust
+#[derive(Debug, Copy, Clone)]
+struct Rectangle {
+    width: u64,
+    height: u64,
+}
+
+impl Rectangle {
+    fn can_hold(&self, other: &Rectangle) -> bool {
+        self.width > other.width && self.height > other.height
+    }
+
+    fn stretch(&self, factor: u64) -> Option<Self> {
+        let w = self.width.checked_mul(factor)?;
+        let h = self.height.checked_mul(factor)?;
+        Some(Rectangle { width: w, height: h })
+    }
+}
+```
+
+In order to prove properties about the rectangle, we wrote a proof harness. The proof harness tried to prove that when the rectangle is stretched, it can hold another rectangle of its original size (dimensions). If proven, this means that for any given stretch factor and any height and width of the original rectangle, this property holds true.
```

Comment:
Shouldn't we refrain from using terms such as "proof" or "prove" throughout the post?

Reply:
We've used "proofs" and "prove" in previous blog posts, so I thought readers would be familiar with those words.

Reply:
Discussed offline.